### PR TITLE
clientv3/integration: add member promote failure test cases

### PR DESCRIPTION
**Pull request by jingyih**
Add 2 failure cases for member promote. "Learner not ready" failure case is already covered in `TestMemberPromote`[1]. This PR completes all 3 cases mentioned in the doc [2].

cc @xiang90 @WIZARD-CXY @gyuho 

Reference:
[1] https://github.com/etcd-io/etcd/blob/2f7121b420960bc6906d64ec853cbd0159c92dff/clientv3/integration/cluster_test.go#L253-L255
[2] [Doc: Error cases when promoting a learner member](https://github.com/etcd-io/etcd/blob/master/Documentation/op-guide/runtime-configuration.md#error-cases-when-promoting-a-learner-member)

### CODE REVIEW

File: clientv3/integration/cluster_test.go [hunk 1]
```diff
@@ -24,6 +24,7 @@ import (
 	"go.etcd.io/etcd/integration"
 	"go.etcd.io/etcd/pkg/testutil"
 	"go.etcd.io/etcd/pkg/types"
+	"math/rand"
```
Comment (xiang90) [inline clientv3/integration/cluster_test.go]: move to line 23

**Comment by xiang90** [comment]
lgtm

File: clientv3/integration/cluster_test.go (hunk 1 above)
Comment (jingyih) [reply clientv3/integration/cluster_test.go]: Thanks. Fixed.

**Comment by WIZARD-CXY** [comment]
lgtm

**Pull request merged**